src/x11/x.c

```diff
@@ -386,13 +386,27 @@ bool x_is_idle(void)
  */
 static void x_handle_click(XEvent ev)
 {
-        if (ev.xbutton.button == Button3) {
+        enum mouse_action act;
+
+        switch (ev.xbutton.button) {
+                case Button1:
+                        act = settings.left_click;
+                        break;
+                case Button2:
+                        act = settings.middle_click;
+                        break;
+                case Button3:
+                        act = settings.right_click;
+                        break;
+        }
+
+        if (act == push_all) {
```

Comment:
There's some undefined behaviour here since `act` is uninitialized by default - are there any other mouse button events? Not sure, but it's best to handle it in any case. (probably ignore the event in the default case)